tests/test_gg.py

```diff
@@ -431,7 +431,7 @@ def test_gg():
     g2 = -gamma0 * np.exp(-r2/2.) * (2.*x*y)/r0**2
 
     cat = treecorr.Catalog(x=x, y=y, g1=g1, g2=g2, x_units='arcmin', y_units='arcmin')
-    gg = treecorr.GGCorrelation(bin_size=0.1, min_sep=1., max_sep=100., sep_units='arcmin',
+    gg = treecorr.GGCorrelation(bin_size=0.1, min_sep=1., max_sep=100., nbins=None, sep_units='arcmin',
```

Comment:
Don't do this.  Leave the old tests alone, since they still need to work the same way they used to.  Please instead add a new test using None to make sure it is valid and does the same thing.

E.g. after what is currently here (in the old version), add:
```
# Using None rather than omiting nbins is equivalent.
gg2 = treecorr.GGCorrelation(bin_size=0.1, min_sep=1., max_sep=100., nbins=None, sep_units='arcmin')
gg2.process(cat)
assert gg2 == gg
```
I believe this would fail in 4.2, but should work with your update.  Likewise for the other tests you changed.

Reply:
Looks like `gg2 == gg` doesn't hold true at machine precision even when they are initialized exactly in the same manner: https://colab.research.google.com/drive/1gXNzl0EOr8zui7luENpeQt74AOI-xPy5?usp=sharing

I'll expand the assertion into approximate comparison of all its attributes listed in `__eq__`.

Reply:
The problem is threading changing the order of operations randomly.  If you do both with num_threads=1, it should come out exact to machine precision.
```
gg2 = treecorr.GGCorrelation(bin_size=0.1, min_sep=1., max_sep=100., nbins=None, sep_units='arcmin')
gg2.process(cat, num_threads=1)  # Make sure OpenMP doesn't change order of operations.
gg.process(cat, num_threads=1)
assert gg2 == gg
```